core/gpslocation.cpp

```diff
@@ -436,13 +436,12 @@ void GpsLocation::deleteFixFromStorage(gpsTracker &gt)
 
 void GpsLocation::deleteGpsFix(qint64 when)
 {
-	struct gpsTracker defaultTracker;
-	defaultTracker.when = 0;
-	struct gpsTracker deletedTracker = m_trackers.value(when, defaultTracker);
-	if (deletedTracker.when != when) {
+	auto it = m_trackers.find(when);
+	if (it == m_trackers.end()) {
 		qDebug() << "can't find tracker for timestamp" << when;
```

Comment:
probably best to include `GpsLocation::deleteGpsFix():` to indicate the origin of the message.
`qWarning()` is probably a better choice too.

Reply:
Is it OK to do this in the same commit?

Reply:
at think yes, as long as it mentions it in the commit message.